Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -646,558 +587,373 @@ protected virtual void Dispose(bool disposing)
-                                    {
-                                        // Keep the number of gateways on this interface, along with its index.
-                                        Tag = ipProperties.GetIPv6Properties().Index
-                                    };
-
-                                    int tag = nw.Tag;
-                                    if (ipProperties.GatewayAddresses.Count > 0 && !nw.IsLoopback())
-                                    {
-                                        // -ve Tags signify the interface has a gateway.
-                                        nw.Tag *= -1;
-                                    }
-
-                                    _interfaceAddresses.AddItem(nw, false);
-
-                                    // Store interface name so we can use the name in Collections.
-                                    _interfaceNames[adapter.Description.ToLower(CultureInfo.InvariantCulture)] = tag;
-                                    _interfaceNames["eth" + tag.ToString(CultureInfo.InvariantCulture)] = tag;
-                                }
-                            }
-                        }
-#pragma warning disable CA1031 // Do not catch general exception types
-                        catch (Exception ex)
-                        {
-                            // Ignore error, and attempt to continue.
-                            _logger.LogError(ex, "Error encountered parsing interfaces.");
-                        }
-#pragma warning restore CA1031 // Do not catch general exception types
-                    }
-                }
-                catch (Exception ex)
+                if (match)
                 {
-                    _logger.LogError(ex, "Error in InitialiseInterfaces.");
+                    break;
                 }
+            }
 
-                // If for some reason we don't have an interface info, resolve our DNS name.
-                if (_interfaceAddresses.Count == 0)
-                {
-                    _logger.LogError("No interfaces information available. Resolving DNS name.");
-                    IPHost host = new IPHost(Dns.GetHostName());
-                    foreach (var a in host.GetAddresses())
-                    {
-                        _interfaceAddresses.AddItem(a);
-                    }
-
-                    if (_interfaceAddresses.Count == 0)
-                    {
-                        _logger.LogWarning("No interfaces information available. Using loopback.");
-                    }
-                }
+            if (!match)
+            {
+                return match;
+            }
 
-                if (IsIP4Enabled)
-                {
-                    _interfaceAddresses.AddItem(IPNetAddress.IP4Loopback);
-                }
+            foreach (var excludedSubnet in _excludedSubnets)
+            {
+                match = match && !excludedSubnet.Contains(address);
```

Comment:
Same as above. Just check the condition and return true instead of storing the result